```
/home/runner/work/MudBlazor/MudBlazor/src/MudBlazor/Components/Dialog/DialogOptions.cs(26,22): error CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. [/home/runner/work/MudBlazor/MudBlazor/src/MudBlazor/MudBlazor.csproj::TargetFramework=net7.0]
```
```cs
    public class DialogOptions
    {
        public DialogPosition? Position { get; set; }

        public MaxWidth? MaxWidth { get; set; }

        public bool? DisableBackdropClick { get; set; }
        public bool? CloseOnEscapeKey { get; set; }
        public bool? NoHeader { get; set; }
        public bool? CloseButton { get; set; }
        public bool? FullScreen { get; set; }
        public bool? FullWidth { get; set; }
        public string? ClassBackground { get; set; }
    }
```

I don't understand, you don't accept the possibility of a null on a string? The parameter may not be present, so it should be null, it is optional.

--- a/src/MudBlazor.Docs/Pages/Components/Dialog/Examples/DialogBlurryExample.razor
+++ b/src/MudBlazor.Docs/Pages/Components/Dialog/Examples/DialogBlurryExample.razor
@@ -17,7 +17,7 @@
 
     private void OpenDialog()
     {
-        var options = new DialogOptions { ClassBackground = "my-custom-class" };
+        var options = new DialogOptions { BackgroundClass = "my-custom-class" };
         DialogService.Show<DialogBlurryExample_Dialog>("Simple Dialog", options);
     }
 }

--- a/src/MudBlazor.Docs/Pages/Components/Dialog/Examples/DialogConfigurationExample.razor
+++ b/src/MudBlazor.Docs/Pages/Components/Dialog/Examples/DialogConfigurationExample.razor
@@ -8,5 +8,5 @@
     NoHeader="true"
     Position="DialogPosition.Center"
     CloseOnEscapeKey="true"
-    ClassBackground="my-custom-class"
+    BackgroundClass="my-custom-class"
 />

--- a/src/MudBlazor/Components/Dialog/DialogOptions.cs
+++ b/src/MudBlazor/Components/Dialog/DialogOptions.cs
@@ -24,7 +24,7 @@ namespace MudBlazor
         public bool? CloseButton { get; set; }
         public bool? FullScreen { get; set; }
         public bool? FullWidth { get; set; }
-        public string? ClassBackground { get; set; }
+        public string? BackgroundClass { get; set; }
     }
 
     public enum DialogPosition

--- a/src/MudBlazor/Components/Dialog/DialogOptions.cs
+++ b/src/MudBlazor/Components/Dialog/DialogOptions.cs
@@ -6,6 +6,7 @@
 // License: MIT
 // See https://github.com/Blazored
 
+#nullable enable
 
 using System.ComponentModel;
 

--- a/src/MudBlazor/Components/Dialog/MudDialogInstance.razor.cs
+++ b/src/MudBlazor/Components/Dialog/MudDialogInstance.razor.cs
@@ -174,7 +174,7 @@ namespace MudBlazor
             DisableBackdropClick = SetDisableBackdropClick();
             CloseOnEscapeKey = SetCloseOnEscapeKey();
             Class = Classname;
-            BackgroundClassname = new CssBuilder("mud-overlay-dialog").AddClass(Options.ClassBackground).Build();
+            BackgroundClassname = new CssBuilder("mud-overlay-dialog").AddClass(Options.BackgroundClass).Build();
         }
 
         private string SetPosition()

--- a/src/MudBlazor/Components/Dialog/MudDialogProvider.razor.cs
+++ b/src/MudBlazor/Components/Dialog/MudDialogProvider.razor.cs
@@ -6,6 +6,8 @@
 // License: MIT
 // See https://github.com/Blazored
 
+#nullable enable
+
 using System;
 using System.Collections.ObjectModel;
 using System.Linq;
@@ -17,8 +19,8 @@ namespace MudBlazor
 {
     public partial class MudDialogProvider : IDisposable
     {
-        [Inject] private IDialogService DialogService { get; set; }
-        [Inject] private NavigationManager NavigationManager { get; set; }
+        [Inject] private IDialogService DialogService { get; set; } = null!;
+        [Inject] private NavigationManager NavigationManager { get; set; } = null!;
 
         [Parameter][Category(CategoryTypes.Dialog.Behavior)] public bool? NoHeader { get; set; }
         [Parameter][Category(CategoryTypes.Dialog.Behavior)] public bool? CloseButton { get; set; }
@@ -88,12 +90,12 @@ namespace MudBlazor
             StateHasChanged();
         }
 
-        private IDialogReference GetDialogReference(Guid id)
+        private IDialogReference? GetDialogReference(Guid id)
         {
             return _dialogs.SingleOrDefault(x => x.Id == id);
         }
 
-        private void LocationChanged(object sender, LocationChangedEventArgs args)
+        private void LocationChanged(object? sender, LocationChangedEventArgs args)
         {
             DismissAll();
         }

--- a/src/MudBlazor/Components/Dialog/MudDialogProvider.razor.cs
+++ b/src/MudBlazor/Components/Dialog/MudDialogProvider.razor.cs
@@ -29,7 +29,7 @@ namespace MudBlazor
         [Parameter][Category(CategoryTypes.Dialog.Appearance)] public bool? FullWidth { get; set; }
         [Parameter][Category(CategoryTypes.Dialog.Appearance)] public DialogPosition? Position { get; set; }
         [Parameter][Category(CategoryTypes.Dialog.Appearance)] public MaxWidth? MaxWidth { get; set; }
-        [Parameter][Category(CategoryTypes.Dialog.Appearance)] public string? ClassBackground { get; set; }
+        [Parameter][Category(CategoryTypes.Dialog.Appearance)] public string? BackgroundClass { get; set; }
 
         private readonly Collection<IDialogReference> _dialogs = new();
         private readonly DialogOptions _globalDialogOptions = new();
@@ -47,7 +47,7 @@ namespace MudBlazor
             _globalDialogOptions.Position = Position;
             _globalDialogOptions.FullWidth = FullWidth;
             _globalDialogOptions.MaxWidth = MaxWidth;
-            _globalDialogOptions.ClassBackground = ClassBackground;
+            _globalDialogOptions.BackgroundClass = BackgroundClass;
         }
 
         protected override Task OnAfterRenderAsync(bool firstRender)

